sdk/src/lib/redeemer-proxy.ts

```diff
@@ -25,6 +26,8 @@ export default class OrangeKitTbtcRedeemerProxy implements TbtcRedeemerProxy {
 
   #sharesAmount: bigint
 
+  #builtDataStepCallback?: BuiltDataStepCallback
```

Comment:
Can't we use `onSignMessageStepCallback`?

Reply:
Unfortunately, no. Withdrawal flow is the following three steps:

1. Preparing transaction
2. Opening wallet for signature
3. Awaiting signature confirmation

So we need to know when the data for the transaction is ready to display step 2. The status change in `onSignMessageStepCallback` matches the change from step 2 to step 3.

Reply:
> Isn't the data ready when we ask for a signature?

Yeah the data is ready at that time.

> Can't we display the preparing transaction step just after invoking the withdrawal function, then change the status in onSignMessageStepCallback to Opening wallet for signature and in messageSignedStepCallback display Awaiting signature confirmation?

The`Opening wallet for signature` step should change to `Awaiting signature confirmation` when the wallet window opens. Therefore, we need the additional state of `dataBuiltStepCallback`. The status change in `onSignMessageStepCallback` indicates that the wallet should be open.